app/src/main/java/com/github/sdp/mediato/ui/SearchFragment.java

```diff
@@ -147,35 +145,37 @@ public void onClick(View view) {
         this.currentHighlightedButton.setPaintFlags(this.currentHighlightedButton.getPaintFlags() | Paint.UNDERLINE_TEXT_FLAG);
     }
 
-    private void SwitchGridOrRecyclerView(View view) {
-        if (view == peopleButton) {
-            this.gridView.setVisibility(View.GONE);
-            this.recyclerView.setVisibility(View.VISIBLE);
+    private void searchAndDisplayResult(String toBeSearched) {
+        if (this.currentCategory == SearchCategory.PEOPLE) {
+            recyclerView.setLayoutManager(new LinearLayoutManager(this.getContext()));
+            recyclerView.setAdapter(new UserAdapter(searchUserViewModel));
```

Comment:
Isn't it possible to create it on fragment creation and to just switch to it when needed. Because now the layoutManager and the UserAdapter are recreated on each input while currentCategory is PEOPLE